tests/dflydev/tests/util/antPathMatcher/AntPathMatcherTest.php

```diff
@@ -48,11 +48,11 @@ public function formatTestMatchesMessage($pattern, $path)
         return 'Testing path "' . $path . '" against pattern "' . $pattern . '"';
     }
 
-    public function testMatchesProvider()
+    public function provideMatches()
     {
         return array(
             array(
-                'com/t?st',
+                'com/t?st.jsp',
```

Comment:
Adding anchors to the regex pattern is certainly a behavioral change that could break code relying on the old behavior, but I can't find any indication that the old behavior conformed to Ant.

https://ant.apache.org/manual/dirtasks.html#patterns appears to be the canonical reference for the pattern syntax, which the Spring docs also allude to. Within a path component, `?` matches exactly one character and `*` matches zero or more. There's no special treatment of file extensions, so I would not expect `t?st` to match `test.jsp` (although `t?st*` could).

Beyond `?` and `*`, the docs introduce the `**` syntax:

> To make things a bit more flexible, we add one extra feature, which makes it possible to match multiple directory levels. This can be used to match a complete directory tree, or a file anywhere in the directory tree. To do this, `**` must be used as the name of a directory. When `**` is used as the name of a directory in the pattern, it matches zero or more directories. For example: `/test/**` matches all files/directories under `/test/`, such as `/test/x.java`, or `/test/foo/bar/xyz.html`, but not `/xyz.xml`.
>
> There is one "shorthand": if a pattern ends with `/` or `\`, then `**` is appended. For example, `mypackage/test/` is interpreted as if it were `mypackage/test/**`.

I believe the "shorthand" above is correctly handled since you do have a test for `com/foo/` matching `com/foo/bar.jsp` and `com/foo/bar/baz.jsp`.